addons/mass_mailing/models/mailing.py

```diff
@@ -1306,35 +1314,90 @@ def _unsubscribe_token(self, res_id, email):
+        """Store an image in an attachement and returns an url"""
+        attachment = self.env['ir.attachment'].create({
+            'datas': b64image,
+            'name': "cropped_image_mailing_{}".format(self.id),
+            'type': 'binary',})
+
+        attachment.generate_access_token()
+
+        return '/web/image/%s?access_token=%s' % (
+            attachment.id, attachment.access_token)
+
+    def _get_image_from_url(self, url):
+        head = requests.head(url, allow_redirects=True, timeout=5)
+        maxsize = int(tools.config.get("import_image_maxbytes", DEFAULT_IMAGE_MAXBYTES))
+        if (
+            int(head.headers.get('Content-Length', 0)) <= maxsize and
+            'image' in head.headers.get('Content-Type', 'text/html')
+        ):
+            return requests.get(url).content
+
     def _convert_inline_images_to_urls(self, body_html):
         """
         Find inline base64 encoded images, make an attachement out of
         them and replace the inline image with an url to the attachement.
         """
-
-        def _image_to_url(b64image: bytes):
-            """Store an image in an attachement and returns an url"""
-            attachment = self.env['ir.attachment'].create({
-                'datas': b64image,
-                'name': "cropped_image_mailing_{}".format(self.id),
-                'type': 'binary',})
-
-            attachment.generate_access_token()
-
-            return '/web/image/%s?access_token=%s' % (
-                attachment.id, attachment.access_token)
-
-        modified = False
         root = lxml.html.fromstring(body_html)
+        modified = False
         for node in root.iter('img'):
             match = image_re.match(node.attrib.get('src', ''))
             if match:
                 mime = match.group(1)  # unsed
                 image = match.group(2).encode()  # base64 image as bytes
 
-                node.attrib['src'] = _image_to_url(image)
+                node.attrib['src'] = self._image_to_url(image)
+                modified = True
+        # handle the images in mso comments
+        for comment in [c for c in root.xpath('//comment()') if mso_re.match(c.text)]:
+            for match in re.findall(r'<img[^>]*src="(data:image/[A-Za-z]+;base64,[^"]*)"', comment.text):
                 modified = True
+                image = re.sub(r'data:image/[A-Za-z]+;base64,', '', match).encode()  # base64 image as bytes
+                comment.text = comment.text.replace(match, self._image_to_url(image))
+
+        if modified:
+            return lxml.html.tostring(root, encoding='unicode')
 
+        return body_html
+
+    def _crop_vml_images(self, body_html):
+        """
+        Find VML v:image elements, crop their source images, make an attachement
+        out of them and replace their source with an url to the attachement.
+        """
+        root = lxml.html.fromstring(body_html)
+        modified = False
+        for comment in root.xpath('//comment()'):
```

Comment:
Do you mean something like `for comment in [c for c in root.iter() if isinstance(c, etree.Comment)]`? Because `iter` takes a tag name so you can't target comments with it.